### ISSUE TITLE
Implement DVO overrides for replica counts

### ISSUE THREAD

**Issue by infinitewarp** [cloudigrade/cloudigrade#1232]
## Summary

As a cloudigrade sysop, I want to override SRE's DVO checks that enforce 3+ replicas for each deployment because some of my deployments (`cloudigrade-beat`, `cloudigrade-listener`) require only 1 replica.

Current related firing issues:
- [CLOUDMETER-9](https://issues.redhat.com/browse/CLOUDMETER-9) deployment_validation_operator_minimum_three_replicas crcs02ue1
- [CLOUDMETER-13](https://issues.redhat.com/browse/CLOUDMETER-13) deployment_validation_operator_minimum_three_replicas crcp01ue1

DVO Dashboards:
- [stage](https://grafana.app-sre.devshift.net/d/dashdotdb/dash-db?orgId=1&var-datasource=dashdotdb-rds&var-cluster=crcs02ue1&var-namespace=cloudigrade-stage)
- [prod](https://grafana.app-sre.devshift.net/d/dashdotdb/dash-db?orgId=1&var-datasource=dashdotdb-rds&var-cluster=crcp01ue1&var-namespace=cloudigrade-prod)

What is DVO? [See Deployment Validation Operator docs in app-interface](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/dvo.md).
> The [Deployment Validation Operator](https://github.com/app-sre/deployment-validation-operator) (DVO) checks deployments and other resources against a curated collection of best practices. Under the hood, it continuously runs the static analysis tool [KubeLinter](https://github.com/stackrox/kube-linter).

What is our problem? From [Steve's comment here](https://docs.google.com/spreadsheets/d/1mgiQA2dG4bnijCTlaBmKC5Ns7-AjMiPVhStQQokWtz0/edit?disco=AAAAYemoBR4):
> DVO checks will fail, so you'll end up needing to add annotations as per the instructions for "specific workloads" […] 
> […] generally you don't want to ignore the check at the app level (app-1.yml)
> This isn't a blocker in the current state, but we will end up checking DVO alerts again in the checkpoint before going to production. I'd still suggest continuing to follow-up on support for this in Clowder.

Where is Clowder's support?
- [2022-06-01 Slack thread](https://coreos.slack.com/archives/C022YV4E0NA/p1654107782438899)
  > Does anyone know if there are any plans for Clowder to support annotations at the deployments or jobs level in clowdapp.yaml? For Insights Compliance, we’d like to disable the minimum_three_replicas alert for one of our jobs but for now it’s not possible to do that without disabling the alert for the entire clowdapp.
- https://github.com/RedHatInsights/clowder/pull/624
- https://github.com/RedHatInsights/clowder/pull/628
- [RHCLOUD-19668 Change kubelinter annotation on Jobs to 'ignore-all'](https://issues.redhat.com/browse/RHCLOUD-19668)
- [2022-06-09 Slack thread](https://coreos.slack.com/archives/C022YV4E0NA/p1654797488417779)
  > teams are getting kubelinter validation alerts since we changed the annotation

See also:
- https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/dvo.md#disable-dvo-checks
- https://docs.google.com/spreadsheets/d/1mgiQA2dG4bnijCTlaBmKC5Ns7-AjMiPVhStQQokWtz0/edit?disco=AAAAYemoBR4

## Acceptance Criteria

- [x] Verify that the DVO alerts are not firing for three-replica-count any more in stage or prod.
- [x] Share knowledge with the team once we know how this works.

## Assumptions and Questions

- The actual work here might be "talk to people and wait" because we need to understand how to affect this change through Clowder to app-interface, etc.

**Comment by abellotti** [comment]
Still no luck with latest clowder, see my 7/29 comment in https://issues.redhat.com/browse/RHCLOUD-20060

**Comment by abellotti** [comment]
After discussion with the platform team, looks like the annotations on the deployment is all that was needed for the Dvo check as per:

![image](https://user-images.githubusercontent.com/5797328/182562529-8e350c3a-a5a1-488f-87f0-34a3d6db382e.png)

Unblocking this issue.

**Comment by infinitewarp** [comment]
https://github.com/cloudigrade/cloudigrade/pull/1246 merged and deployed to stage and prod, but the annotations aren't showing up presumably because Clowder has not been upgraded in those environments. Maybe this issue is `blocked` again…

**Comment by infinitewarp** [comment]
I heard that an updated Clowder was deployed; so, I checked the DVO complaints.

![2022-08-08 at 4 12 35 PM-prod](https://user-images.githubusercontent.com/1472326/183506481-afccb72e-6b76-472d-b0c1-b2c129d78f78.png)
![2022-08-08 at 4 12 42 PM-stage](https://user-images.githubusercontent.com/1472326/183506490-d79b2ef8-e89f-4ece-b90e-379bd7972aaa.png)

Much better! 😀 Yes, stage still has a those two errors related to the smoke tests…

**Comment by infinitewarp** [comment]
Latest discussion in Slack: https://coreos.slack.com/archives/C023YSA47A4/p1659996991032009

**Comment by infinitewarp** [comment]
Since the original ACs have been met, we're not holding this issue any longer waiting on response about the IQE smoke stage job. We will follow up with that in a separate issue.
